src/java/com/rapleaf/jack/LazyLoadPersistence.java

```diff
@@ -0,0 +1,42 @@
+package com.rapleaf.jack;
+
+public abstract class LazyLoadPersistence<T extends IModelPersistence, D extends GenericDatabases> {
+
+  private final BaseDatabaseConnection conn;
+  private final D databases;
+
+  private volatile T persistence;
+
+  private boolean disableCaching;
+
+  public LazyLoadPersistence(BaseDatabaseConnection conn, D databases) {
+    this.conn = conn;
+    this.databases = databases;
+
+    this.persistence = null;
+    this.disableCaching = false;
+  }
+
+  public T get() {
+    if (persistence == null) {
+      synchronized (this) {
```

Comment:
That was not the goal. Should I be worried about that case?

The goal here was to make sure that we don't create two instances if the same persistence since it will be created at usage and not at init time like before.